src/highlight/highlightUtil.ts

```diff
@@ -18,6 +25,45 @@ export function getHighlightArea(shapes: Rect[]): number {
     return shapes.reduce((area, { height, width }) => area + height * width, 0);
 }
 
+export function getRange(): Range | null {
+    const selection = window.getSelection();
+    if (!selection || selection.type !== 'Range' || selection.isCollapsed) {
+        return null;
+    }
+
+    return selection.getRangeAt(0);
+}
+
+export const DOMRectToRect = ({ height, width, x, y }: DOMRect): Rect => ({
+    height,
+    type: 'rect',
+    width,
+    x,
+    y,
+});
+
+export function getSelectionItem(): SelectionItem | null {
+    const range = getRange();
+
+    if (!range) {
+        return null;
+    }
+
+    const startPage = getPageNumber(range.startContainer as Element);
+    const endPage = getPageNumber(range.endContainer as Element);
+
+    if (!startPage || !endPage || startPage !== endPage) {
+        return null;
+    }
+
+    const boundingRect = DOMRectToRect(range.getBoundingClientRect());
```

Comment:
Do these need to be separate variables if they're only used to define object properties?